core/src/main/java/io/parsingdata/metal/data/ConstantSource.java

```diff
@@ -34,16 +34,16 @@ public ConstantSource(final byte[] data) {
     }
 
     @Override
-    protected byte[] getData(final long offset, final BigInteger length) {
+    protected byte[] getData(final BigInteger offset, final BigInteger length) {
         if (!isAvailable(offset, length)) { throw new IllegalStateException("Data to read is not available ([offset=" + offset + ";length=" + length + ";source=" + this + ")."); }
         final byte[] outputData = new byte[length.intValue()];
-        System.arraycopy(data, (int)offset, outputData, 0, outputData.length);
+        System.arraycopy(data, offset.intValue(), outputData, 0, outputData.length);
```

Comment:
This isn't a problem here, but for future reference, a quote from the `BigInteger#intValue()` JavaDoc:

> if this BigInteger is too big to fit in an int, only the low-order 32 bits are returned

Reply:
As discussed, replaced all invocations of `byteValue()`, `intValue()` and `longValue()` with their `*Exact()` counterparts.